### PR TITLE
Image handling from CR and some enhancements

### PR DESCRIPTION
**Pull request by amdabhad**
- Added support to use images from CR if specified, otherwise use default images
- Fixed ImagePullPolicies for pods
- Removed redundant sidecars field and made changes to use images from existing fields
- Moved code related to sidecars as a separate list of fields, under commented sections
- Cleaned up extra pods specs and few other things
- Also added LivenessProbe, Resizer and ResizerNodeselector

### CODE REVIEW

File: controllers/config/constants.go [hunk 1]
```diff
@@ -15,6 +15,25 @@ const (
 	ENVNodeName    = "NODE_NAME"
 	ENVKubeVersion = "KUBE_VERSION"
 
+	//  Default images for containers
```
Comment (nitishkumar4) [inline controllers/config/constants.go:18]: **Query:**  Are default images constant? Based on tags it seems they are likely to change with version/tag change. So are we planning to update default image under this file when ever  new image is build with new tags ?

Comment (amdabhad) [reply controllers/config/constants.go:18]: The default images here will be overwritten if the images are provided in CR. But I think most of the customers would use the default configuration (where they don't need to know the image names and tags), so yes, we will have to update the images here whenever there is change.

File: controllers/syncer/csi_node.go [hunk 2]
```diff
@@ -342,42 +339,30 @@ func (s *csiNodeSyncer) ensureVolumes() []corev1.Volume {
...
-}
-
-func (s *csiNodeSyncer) getCSINodeDriverRegistrarPullPolicy() corev1.PullPolicy {
-	sidecar := s.getSidecarByName(config.CSINodeDriverRegistrar)
-	if sidecar != nil && sidecar.ImagePullPolicy != "" {
-		return sidecar.ImagePullPolicy
-	}
-	return corev1.PullIfNotPresent
-}
+func (s *csiNodeSyncer) getImage(name string) string {
```
Comment (nitishkumar4) [inline controllers/syncer/csi_node.go:344]: let's add a doc string, stating the purpose of the method and return type.

File: controllers/syncer/csi_syncer.go [hunk 3]
```diff
@@ -631,81 +575,36 @@ func (s *csiControllerSyncer) ensureVolumes() []corev1.Volume {
...
-	return "us.gcr.io/k8s-artifacts-prod/sig-storage/csi-snapshotter:v4.1.1"
-	//return s.getSidecarImageByName(config.CSISnapshotter)
-}
-
-func (s *csiControllerSyncer) getCSIResizerImage() string {
-	return "us.gcr.io/k8s-artifacts-prod/sig-storage/csi-resizer:v1.3.0"
-	//return s.getSidecarImageByName(config.CSIResizer)
-}
+func (s *csiControllerSyncer) getSidecarImage(name string) string {
+	logger := csiLog.WithName("getSidecarImage")
```
Comment (nitishkumar4) [inline controllers/syncer/csi_syncer.go:581]: let's add a doc string, stating the purpose of the method and return type.

File: controllers/syncer/csi_node.go (hunk 2 above)
Comment (amdabhad) [reply controllers/syncer/csi_node.go:344]: I had added the description before the control logic, removed and added improved doc/description before method definition, which I agree is a better place for this. Thanks for pointing this out.

File: controllers/syncer/csi_syncer.go (hunk 3 above)
Comment (amdabhad) [reply controllers/syncer/csi_syncer.go:581]: Same as above, Thanks!

**Review by nitishkumar4 — APPROVED** [review]
LGTM!

File: controllers/syncer/csi_node.go [hunk 4]
```diff
@@ -342,42 +339,31 @@ func (s *csiNodeSyncer) ensureVolumes() []corev1.Volume {
...
-		return sidecar.ImagePullPolicy
+	image := ""
+	switch name {
+	case config.CSINodeDriverRegistrar:
+		if len(s.driver.Spec.DriverRegistrar) != 0 {
+			image = s.driver.Spec.DriverRegistrar
+		} else {
+			image = s.driver.GetDefaultImage(name)
+		}
+	case config.GetNameForResource(config.CSINode, s.driver.Name):
```
Comment (madhuthorat) [inline controllers/syncer/csi_node.go]: Can we simplify this ? Call csiNodeDriver := config.GetNameForResource(config.CSINode, s.driver.Name) before the switch case and then use that here as "case csiNodeDriver"

File: controllers/syncer/csi_node.go [hunk 5]
```diff
@@ -342,42 +339,31 @@ func (s *csiNodeSyncer) ensureVolumes() []corev1.Volume {
...
+		if len(s.driver.Spec.SpectrumScale) != 0 {
+			image = s.driver.Spec.SpectrumScale
+		} else {
+			image = s.driver.GetDefaultImage(config.CSINodeDriverPlugin)
+		}
+	case config.LivenessProbe:
+		image = config.LivenessProbeImage
```
Comment (madhuthorat) [inline controllers/syncer/csi_node.go]: Please add condition to check if s.driver.Spec.Livenessprobe !=0 and then use image specified from CR.

**Review by madhuthorat — APPROVED** [review]
LGTM, Please take care of the 2 comments. Also for nodeContainerHealthPort set default as 9821, which is set since recent v2.3.1 release.

File: controllers/syncer/csi_node.go (hunk 4 above)
Comment (amdabhad) [reply controllers/syncer/csi_node.go]: Done.

File: controllers/syncer/csi_node.go (hunk 5 above)
Comment (amdabhad) [reply controllers/syncer/csi_node.go]: Done.

**Pull request merged**